test/syscalls/linux/pty.cc

```diff
@@ -387,6 +387,18 @@ PosixErrorOr<size_t> PollAndReadFd(int fd, void* buf, size_t count,
 }
 
 TEST(PtyTrunc, Truncate) {
+  // Make sure we're ignoring SIGHUP, which will be sent to this process once we
+  // disconnect they TTY.
+  struct sigaction sa = {};
+  sa.sa_handler = SIG_IGN;
+  sa.sa_flags = 0;
+  sigemptyset(&sa.sa_mask);
+  struct sigaction old_sa;
+  ASSERT_THAT(sigaction(SIGHUP, &sa, &old_sa), SyscallSucceeds());
```

Comment:
Please use the ScopedSigaction helper for this. It takes care of the Cleanup for you:
https://github.com/google/gvisor/blob/108410638aa8480e82933870ba8279133f543d2b/test/util/signal_util.cc#L79